### PR TITLE
refactor(cheatcode): use rate limit args in create fork cheatcode

### PR DESCRIPTION
**Pull request by gr4yha7**
## Motivation
Ref: #5885.

### CODE REVIEW

File: crates/evm/src/executor/fork/multi.rs [hunk 1]
```diff
@@ -476,13 +465,11 @@ fn create_fork_id(url: &str, num: Option<u64>) -> ForkId {
...
     mut fork: CreateFork,
-    retries: u32,
-    backoff: u64,
 ) -> eyre::Result<(CreatedFork, Handler)> {
     let provider = Arc::new(
         ProviderBuilder::new(fork.url.as_str())
-            .max_retry(retries)
-            .initial_backoff(backoff)
+            .max_retry(fork.evm_opts.fork_retries.unwrap())
+            .initial_backoff(fork.evm_opts.fork_retry_backoff.unwrap())
```
Comment (mattsse) [inline crates/evm/src/executor/fork/multi.rs:472]: these unwraps are unsafe,

I think we can add additional builder functions that take an option for convenience.

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit

File: crates/evm/src/executor/fork/multi.rs (hunk 1 above)
Comment (gr4yha7) [reply crates/evm/src/executor/fork/multi.rs:472]: okay, i'll work on it.

**Comment by Evalir** [comment]
Closing in favor of #6193 — thanks!

**Pull request closed without merging**